carma_wm/src/WMListenerWorker.cpp

```diff
@@ -274,67 +296,67 @@ void WMListenerWorker::mapUpdateCallback(const autoware_lanelet2_msgs::MapBinPtr
   */
 void WMListenerWorker::newRegemUpdateHelper(lanelet::Lanelet parent_llt, lanelet::RegulatoryElement* regem) const
 {
-
-  auto factory_pcl = lanelet::RegulatoryElementFactory::create(regem->attribute(lanelet::AttributeName::Subtype).value(),
+  auto factory_regem = lanelet::RegulatoryElementFactory::create(regem->attribute(lanelet::AttributeName::Subtype).value(),
                                                             std::const_pointer_cast<lanelet::RegulatoryElementData>(regem->constData()));
 
   // we should extract general regem to specific type of regem the geofence specifies
   switch(resolveGeofenceType(regem->attribute(lanelet::AttributeName::Subtype).value()))
   {
     case GeofenceType::PASSING_CONTROL_LINE:
     {
-
-      lanelet::PassingControlLinePtr control_line = std::dynamic_pointer_cast<lanelet::PassingControlLine>(factory_pcl);
+      lanelet::PassingControlLinePtr control_line = std::dynamic_pointer_cast<lanelet::PassingControlLine>(factory_regem);
```

Comment:
Should have caught this earlier, but if a dynamic_pointer_cast is being used you should check the resulting pointer is populated as dynamic_cast returns a nullptr if the cast failed.

Reply:
added guards for every one of them